corehq/apps/ota/views.py

```diff
@@ -64,7 +64,11 @@ def search(request, domain):
         search_es = search_es.case_property_query(key, value, fuzzy=(key in fuzzies))
     results = search_es.values()
     if should_use_sql_backend(domain):
-        cases = [get_instance_from_data(CommCareCaseSQLSerializer, sqlify_result(result)) for result in results]
+        # This is always going to fail validation, because case_id is marked as unique, and the instance will
+        # always already exist:
+        # cases = [get_instance_from_data(CommCareCaseSQLSerializer, sqlify_result(result)) for result in results]
+        # Ugly DB hit, mitigated only by CASE_SEARCH_MAX_RESULTS:
+        cases = CommCareCaseSQL.objects.filter(case_id__in=[result['_id'] for result in results])
```

Comment:
CommCareCase.wrap() works great.

(Along the way I also tried a dict whose keys can be addressed as attributes: 

    class AttrDict(dict):
        """
        A dict whose keys you can get like attributes.
    
        >>> breakfast = AttrDict(egg=1)
        >>> breakfast['egg']
        1
        >>> breakfast.egg
        1
    
        """
        def __getattr__(self, key):
            try:
                return super(AttrDict, self).__getattribute__(key)
            except AttributeError as err:
                try:
                    return self[key]
                except KeyError:
                    raise err

but it can't mimic a model because it doesn't change the data types of dates.)